lib/node/http.js

```diff
@@ -383,6 +385,108 @@ class HTTP extends Server {
 
       res.json(200, { success: true });
     });
+
+    this.get('/info/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid parameter.');
+
+      const network = this.network;
+      const height = this.chain.height;
+      const nameHash = rules.hashName(name);
+      const reserved = rules.isReserved(nameHash, height + 1, network);
+      const [start, week] = rules.getRollout(nameHash, network);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      let info = null;
+
+      if (ns) {
+        if (!ns.isExpired(height, network))
+          info = ns.getJSON(height, network);
+      }
+
+      return res.json(200, {
+        start: {
+          reserved: reserved,
+          week: week,
+          start: start
+        },
+        info
+      });
+    });
+
+    this.get('/name/hash/:hash', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const hash = valid.bhash('hash');
+
+      if (!hash)
+        throw new Error('Invalid hash.');
+
+      const ns = await this.chain.db.getNameState(hash);
+
+      if (!ns)
+        return res.json(404);
+
+      return res.json(200, { name: ns.name.toString('binary') });
+    });
+
+    this.get('/resource/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid name.');
+
+      const nameHash = rules.hashName(name);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      if (!ns || ns.data.length === 0)
+        return null;
+
+      const resource = Resource.decode(ns.data);
+
+      return res.json(200, resource.toJSON());
```

Comment:
Pass in the name to `getJSON`, see here:
https://github.com/handshake-org/hsd/blob/696dfc4601ef5e8cd954cde727142f39c965fcf6/lib/dns/resource.js#L1309